hack/validate/vendor

```diff
@@ -3,10 +3,15 @@
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source "${SCRIPTDIR}/.validate"
 
+go_mod_tidy(){
+	${SCRIPTDIR}/../go-mod-prepare.sh
+	GO111MODULE=auto go mod tidy -modfile 'vendor.mod' -compat 1.17
+}
+
 validate_vendor_diff(){
 	IFS=$'\n'
 	# shellcheck disable=SC2207
-	files=( $(validate_diff --diff-filter=ACMR --name-only -- 'vendor.conf' 'vendor/' || true) )
+	files=( $(validate_diff --diff-filter=ACMR --name-only -- 'vendor.sum' 'vendor.mod' 'vendor/' || true) )
```

Comment:
@tonistiigi thanks for checking this, well spotted! I have looked into this and it doesn't seem to detect uncommitted changes to any of these files, it seems to be just getting a list of files that changed between `origin/HEAD` and `HEAD`.

When I ran `bash -x hack/validate/vendor`, the command it called was this:
```
+ git diff 088afc99e4bf8adb78e29733396182417d67ada2...0089b1ffb718e9c413a0687dd714dfc5c0829cb4 --diff-filter=ACMR --name-only -- vendor.sum vendor.mod vendor/
```
Where 088afc99e4bf8adb78e29733396182417d67ada2 is what's in master right now, and 0089b1ffb718e9c413a0687dd714dfc5c0829cb4 is the head of my branch.

Reply:
Doesn't at least line 21 need `vendor.mod` and `vendor.sum` as well?

Reply:
> Doesn't at least line 21 need `vendor.mod` and `vendor.sum` as well?

You are right, that looks like where detection of uncommitted changes happens.

I can also see what `validate_diff` is all about, it checks what files changed in `HEAD` vs `origin/HEAD`.
I just tend to use `git diff` to check for uncommitted changes, so I too quickly assumed that's what `validate_diff` might be for, the function name isn't very specific...

Reply:
I've updated this script just now.